### PR TITLE
POS3-108: Sign logged Messages with UUID

### PR DESCRIPTION
**Pull request by momosh-ethernal**
We needed a way to track messages that are being propagated with the GossipSub gateway mechanism. Now each message contains a UUID that will be used as a log field.

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -17,18 +20,27 @@ import (
...
+	SendMessage(size int) error
+	Stop() error
+	NumPeers() int
+}
+
+type GossipAgent struct {
 	Host   host.Host
 	Logger *zap.Logger
-	Config *AgentConfig
+	Config *GossipConfig
```
Comment (ferranbt) [inline agent/agent.go:39]: Could we address variable changes in another PR later on? It is hard to follow the changes, much of the diff lines are changing variable names.

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: agent/agent.go (hunk 1 above)
Comment (momosh-ethernal) [reply agent/agent.go:39]: sure thing 👍

**Review by 0xSasaPrsic — APPROVED** [review]
